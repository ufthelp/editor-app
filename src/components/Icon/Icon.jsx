import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import './Icon.scss';

const Icon = ({ iconImage, size, color }) => (
    <div className="iconCta">
        <FontAwesomeIcon icon={iconImage} size={size} color={color}/>
    </div>
  );

  Icon.propTypes = {
    iconImage: PropTypes.string.isRequired,
    size: PropTypes.string,
    color : PropTypes.string
}

Icon.defaultProps = { 
    size: '2x',
    color: 'gray'
 };

export default Icon;
