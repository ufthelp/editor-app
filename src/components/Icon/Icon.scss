.iconCta {
  cursor: pointer;
  svg:hover{
    color: white;
  }
}