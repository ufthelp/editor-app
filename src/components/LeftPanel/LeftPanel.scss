@import "../../scss/variables.scss";

.leftPanelCta{
    background-color: $defaultPanelbgColor;
    width: $leftPanelWidth;
    > div {
      flex: 1;   
      display: flex;
      flex-direction: column;
      justify-content: center;
      padding: 1rem;
      cursor: pointer;
      align-items: center;
    }
  }