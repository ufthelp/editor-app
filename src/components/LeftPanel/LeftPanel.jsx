import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './LeftPanel.scss';
import Icon from '../../UI/Icon';

class LeftPanel extends Component {
    static propTypes = {
        handleShowRightView: PropTypes.func.isRequired,
        selectedIndex: PropTypes.number.isRequired
    }

    render() {
        //icon types 
        const arrIconImages = ['file','search','code-branch'];
        const {handleShowRightView,selectedIndex} = this.props;
        
        return (
            <div className="leftPanelCta">
                {
                    arrIconImages.map((icon, idx) =>  
                        <div onClick={() => handleShowRightView(icon,idx)} key={idx}>
                            <Icon  
                                iconImage={icon} 
                                color={selectedIndex === idx ? 'white':''}
                            /> 
                        </div>
                    )
                }               
            </div>
        );
    }
}

export default LeftPanel;
