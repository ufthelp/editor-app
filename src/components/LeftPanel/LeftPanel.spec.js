import React from 'react';
import {
    shallow,
} from "enzyme";
import LeftPanel from './LeftPanel';
import Icon from '../Icon';

describe('<LeftPanel />', () => {
            const handleClick = jest.fn();
            let wrapper;
            beforeEach(() => wrapper = shallow( 
                < LeftPanel 
                    handleShowRightView={handleClick} 
                    selectedIndex={1}
                /> ));

            it('should render correctly', () => expect(wrapper).toMatchSnapshot());

            it('should contain 3 <Icon /> component',()=>{
               expect(wrapper.find(Icon).length).toBe(3);
            })
})
