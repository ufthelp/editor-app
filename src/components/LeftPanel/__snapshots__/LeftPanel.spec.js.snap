// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftPanel /> should render correctly 1`] = `ShallowWrapper {}`;
